lib/job/arFindingAidJob.class.php

```diff
@@ -22,41 +22,67 @@
  * @author     Mike G <mikeg@artefactual.com>
  */
 
-class arGenerateFindingAidJob extends arBaseJob
+class arFindingAidJob extends arBaseJob
 {
   /**
    * @see arBaseJob::$requiredParameters
    */
   protected $extraRequiredParameters = array('objectId');
 
-  private $resourceId = 0;
+  private $resource = null;
 
   public function runJob($parameters)
   {
-    $appRoot = rtrim(sfConfig::get('sf_root_dir'), '/');
-
-    $this->resourceId = $parameters['objectId'];
-    $resource = QubitInformationObject::getById($this->resourceId);
+    $this->resource = QubitInformationObject::getById($parameters['objectId']);
 
-    if (!$resource)
+    if (!$this->resource)
     {
-      $this->error('Error: Could not find an information object with id=' . $this->resourceId);
+      $this->error('Error: Could not find an information object with id=' . $this->resource->id);
+
       return false;
     }
 
     $this->checkDownloadsExistsAndCreate();
-    $this->info("Generating finding aid ({$resource->slug})...");
+
+    if (isset($parameters['delete']) && $parameters['delete'])
+    {
+      $this->delete();
+    }
+    else if (isset($parameters['tempPath']))
+    {
+      $this->upload($parameters['tempPath']);
+    }
+    else
+    {
+      $this->generate();
+    }
+
+    // Update resource in ES to add finding aid transcript
+    QubitSearch::getInstance()->update($this->resource);
```

Comment:
If `true === $parameters['delete']` could we just delete the stuff from the document that needs to be deleted instead of repopulate? I know that we normally take the brute-force path but this could be an opportunity to start working on granular updates and perhaps implement some function helpers that enable us to do that from different places in the application?

I've found this in the ES Update API:

```
curl -XPOST 'localhost:9200/test/type1/1/_update' -d '{
    "script" : "ctx._source.remove(\"name_of_field\")"
}'
```

Is that what we need? Can we do that from Elastica?